### ISSUE TITLE
[SUGGESTION]: Add a solution for "quick vitamins"

### ISSUE THREAD

**Issue by Symi001** [pokeclicker/pokeclicker#4020]
### Current Behavior

We currently have 3 different vitamin types, and it now takes a very long time to "optimally" set your poke vitamins. 
We used to be able to do it "quickly" with proteins by simply adding the Max amount each time you ventured into a new region, which was "simple and quick" (still took time ofc, but it was quite different).

### Improved Behavior

I don't have a definite UX or UI proposal for it, but I'd imagine a solution to setup your poke to the optimal possible value, with a desclaimer that it'd not take into account the region debuff (? If we want it?).
Ofc, this solution (a button in the current panel maybe?) would not work if the player doesn't have the vitamins to use. The player would still have to go through their **whole** list by hand. but at least they can use the optimal vitamins in one click per poke, instead of a lot more. 

### Reason for change

First, I'm lazy, and I think a lot of ppl are :)
Second, I truly think the vitamins will benefit from this, because I'm afraid ppl will never bother using vitamins on less efficient poke because of the current hassle it could represent. 
Let's say you're a fresh player, getting to Unova. Your previous top pokemon was Heatran with 20 calcium. Now in Unova, you'll need to leave 5 Calcium, and use 20 Carbos, that's quite a big number of clicks. Then onward to Kalos, 7 Calcium, 23 Carbos. And this is all for one poke. Makes quite a lot of clicks to me.

### Detailed Description

I hope you'll find enough details in the previous paragraphs so I'll stick to a list sum up

Have a system that allows to quickly fill the vitamins of a poke to their optimal values with the constraints:
1.  Fails if the player does not have the necessary amount of vitamins.
2. Takes into account the current region (obvious, I know ^^).
3. [OPTIONAL] Takes region debuff into account in a separate setting.

Thank you for reading, and don't hesitate to ping me in Discord if you want to talk about this ;)

**Comment by DraKay** [comment]
Regional debuff don't affect the vitamin set up

**Comment by Symi001** [comment]
That... is completely true. Sry ^^"

**Comment by whatIsInAUsername** [comment]
Feel free to point out if I've overlooked something, as I more than likely have, but it feels like the following would work well here.

- A toggle for auto-allocating vitamins on all mons (Maybe to the left or right of the All "Vitamins" button in the "Item Bag" tab)
  - Toggle could be unlocked upon reaching Unova or Kalos
  - Toggle could default to unchecked (**Do not** auto-allocate)
- A toggle to prohibit auto-allocating vitamins at the far right of the row for each mon on the Vitamin allocation pages
  - Toggle could default to unchecked (**Do** allow auto-allocate)
  - Toggle could automatically become checked upon adding or subtracting a vitamin from a mon
- An explainer at the unlock region should be added but isn't strictly be necessary
- A "Hide Pokémon set to auto-allocate" filter could be made available on the Vitamin allocation pages

## Basics
```
if the overall auto-allocate toggle is unchecked:
  vitamin allocation functions as it does now
elif the overall auto-allocate toggle is checked:
  On purchase, vitamins are auto-allocated to mons that [
    have their individual prohibit toggle set to off,
    have not reached their optimal vitamin count for the given vitamin and current region
  ]
  # Vitamins could be assigned in whatever order the dev team sees fit.
  # I dunno, maybe lowest breeding efficiency first? I'm begging, not choosing here.
```

```
if the user has unallocated vitamins and manually allocates a vitamin to a mon:
  the vitamin is decremented from the unallocated
if the user has **no** unallocated vitamins and manually allocates a vitamin to a mon:
  the vitamin is deallocated from a mon set to auto-allocate in the reverse order of allocation
```

## Might be Nice

- Probably checking the overall auto-allocate toggle should trigger an immediate allocation of all unallocated vitamins **and** reallocation of suboptimal vitamins allocated to mons with prohibit toggle unchecked
- Button to deallocate all vitamins from all mons with prohibit toggle unchecked, but only if the overall auto-allocate toggle is unchecked, of course
- I'm unsure of what the behavior should be upon catching a new mon
  - Deallocating from mons until the caught mon reaches parity feels correct but weird
  - Only allocating newly purchased vitamins feels weird too (What do I know, though?)

## Not Necessary

- Could allow filters and allocation order to be set by the user, but I don't feel that it's strictly necessary (If the user is pursuing a more specific meta, let them take pride in it and set those bad boys by hand)

**Comment by whatIsInAUsername** [comment]
First of all, your paint markup is superb, I love it.

That makes sense and I totally get not wanting to automate the entire game away. I just always look at it as finding ways to let players play the game they want to even to the detriment of how I intended them to play it. Moving forward I hope we'll more games that build to that ethos. If players find fine-grained micro gratifying I want them to be able to do that and if some players want to set it and forget it then I think that'd be lovely too.

I figure you still have a trade-off there. Players that take the time to set vitamins manually probably have a specific meta they're pursuing and those that don't are fine progressing slower if it means not burning real-life time on allocating vitamins. Neither strategy is given more resources; we're just letting them choose what they want to do with them.

Ultimately, I'd be happy with the buttons as you envisioned them too.

**Comment by RegisCoaxans** [comment]
That button is very unlikely to happen. At most, we could agree on easing the process of attributing vitamins further more.
What you suggested, though, would definitely make so people do not care about the feature, they would click "Use opti" and ignore whatever the vitamins are and do. There would not have been a vitamin rework if we only wanted such a blind use.

**Comment by whatIsInAUsername** [comment]
> That button is very unlikely to happen. At most, we could agree on easing the process of attributing vitamins further more. What you suggested, though, would definitely make so people do not care about the feature, they would click "Use opti" and ignore whatever the vitamins are and do. There would not have been a vitamin rework if we only wanted such a blind use.

ok

**Comment by goku90504** [comment]
#4035 was a better suggestion IMO but I'd be happy if we could just type a number in each box instead of having to click in the stupidly limited 1/5/10/max options we've got other boxes where we type things in in the game why not for the vitamin amounts as well? even if just a custom multi instead of being limited to 1/5/10 or even a hot key to shift between 1/5/10 without having to move the mouse all the way to the top and back again

**Comment by whatIsInAUsername** [comment]
I agree, even being able to upload a csv would be preferable to the way it is now.
Honestly though, if I get some free time in the next week I might clean up what I have for this and see if we can get it merged in.
